File: AA_15-2/cf-349-B.cpp
// Codeforces - Color the Fence - round 202 div 2 B

#include <bits/stdc++.h>

using namespace std;
const long int MAX = 100001;

int main() {
	long int v;
	cin >> v;

	vector<long int> ad;
	long int menor = MAX;
	long int menorIndex = -1;
	for (int i = 0; i < 9; i++) {
		long int x;
		cin >> x;

		ad.push_back(x);

		if (menor > x) menor = x;
	}

	if (menor > v) cout << -1 << endl;

	else {
		string loveNumber = '';
		long int qtd = v / menor;

		for (long int i = 0; i < qtd; i++) {
			loveNumber += (string) ad.at(i);
		}

		cout << loveNumber << endl;
	}

	return 0;
}

File: AA_15-2/cf-349-B.py
# Codeforces round 202 div 2 B - Color the Fence

MAX = 1000002

v = int(raw_input())
ad = map(int, raw_input().split())

ans = ''
tentados = 8
while tentados >= 0 and v > 0:
	if (ad[tentados] < v):
		qtd = ad[tentados] / v
		ans += str(tentados + 1) * qtd
		v -= (ad[tentados] / v)
	tentados -= 1

if ans != '':
	print ans

else:
	print -1
